nemo/lightning/pytorch/strategies.py

```diff
@@ -208,6 +226,23 @@ def __init__(
         else:
             raise ValueError(f"Invalid DDP type: {ddp}")
 
+        if self.tp_comm_overlap:
+            if (
+                self.tensor_model_parallel_size < 2
+                or not self.sequence_parallel
+                or not HAVE_TE
+                or not transformer_engine.pytorch.cpp_extensions.userbuf_comm_available()
+            ):
+                logging.warning("Disabling tensor parallel overlap due to incompatible setup.")
```

Comment:
Helpful to print the conditional that failed, so users have a direct remedy from the log.